xlators/cluster/ec/src/ec-inode-write.c

```diff
@@ -710,13 +710,29 @@ ec_setxattr(call_frame_t *frame, xlator_t *this, uintptr_t target,
     ec_cbk_t callback = {.setxattr = func};
     ec_fop_data_t *fop = NULL;
     int32_t error = ENOMEM;
+    data_t *dict_data = NULL;
+    char *mask_key = "user.readmask";
+    int ret = 0;
+    int op_ret = -1;
 
     gf_msg_trace("ec", 0, "EC(SETXATTR) %p", frame);
 
     VALIDATE_OR_GOTO(this, out);
     GF_VALIDATE_OR_GOTO(this->name, frame, out);
     GF_VALIDATE_OR_GOTO(this->name, this->private, out);
 
```

Comment:
check ec_gf_getxattr() on how trusted.ec.heal is handled. This should be something simiar.